src/Message/MakePaymentRequest.php

```diff
@@ -0,0 +1,107 @@
+ * - invoiceId: An optional parameter of the ID of the Invoice against which the payment is to be 
+ * made. If not specified, the oldest unpaid invoice for this AutoBill will be selected for payment. 
+ *
+ * Example:
+ * <code>
+ *   // first create gateway, plan, product and subscription as described in FetchSubsciptionRequest
+ *   // then use subscriptionId returned from $subscriptionResponse->getSubscriptionId(),
+ *   // the payment method id from payment method record and the amount due
+ *
+ *   // get $subscriptionResponse from createSubscription call
+ *
+ *   $makePaymentResponse = $gateway->makePayment(array(
+ *       'subscriptionId' => $subscriptionResponse->getSubscriptionId(), // could also do by reference
+ *       'paymentMethodId' => 'cc-123456' // could also do by reference (VID),
+ *       'amount' => 59.95
+ *   ))->send();
+ *
+ *   if ($makePaymentResponse->isSuccessful()) {
+ *       echo 'Payment summary " . $makePaymentResponse->getSummary() . PHP_EOL;';
+ *       vardump($makePaymentResponse->getTransaction());
+ *   }
+ *
+ * </code>
+ */
+class MakePaymentRequest extends AbstractRequest
+{
+    /**
+     * The name of the function to be called in Vindicia's API
+     *
+     * @return string
+     */
+    protected function getFunction()
+    {
+        return 'makePayment';
+    }
+
+    /**
+     * @return string
+     */
+    protected function getObject()
+    {
+        return self::$SUBSCRIPTION_OBJECT;
+    }
+
+    public function getData()
+    {
+        $subscriptionId = $this->getSubscriptionId();
+        $subscriptionReference = $this->getSubscriptionReference();
+        if (!$subscriptionId && !$subscriptionReference) {
+            throw new InvalidRequestException(
+                'Either the subscriptionId or subscriptionReference parameter is required.'
+            );
+        }
+
+        $paymentMethodId = $this->getPaymentMethodId();
+        $paymentMethodReference = $this->getPaymentMethodReference();
+        if (!$paymentMethodId && !$paymentMethodReference) {
+            throw new InvalidRequestException(
+                'Either the paymentMethodId or paymentMethodReference parameter is required.'
+            );
+        }
+
+        $amount = $this->getAmount();
+        if (!$amount) {
+            throw new InvalidRequestException('The amount parameter is required.');
+        }
+
+        $subscription = new stdClass();
+        $subscription->merchantAutoBillId = $subscriptionId;
+        $subscription->VID = $subscriptionReference;
+
+        $data = array(
+            'action' => $this->getFunction()
+        );
+
+        $data['autobill'] = $subscription;
+        $data['paymentMethod'] = $this->buildPaymentMethod(null);
+        $data['amount'] = $amount;
+        $data['currency'] = null;
+        $data['invoiceId'] = $this->getInvoiceId();
+        $data['overageDisposition'] = null;
+        $data['note'] = 'Payment initiated by makePayment';
+
```

Comment:
i believe we need to add support for the usePaymentMethodForFutureBilling parameter, which appears to not be documented. but it's in the PHP client library and we use it.

```php
    /**
     * makePayment
     *
     * makes a payment against an AutoBill
     *
     * @param string $srd
     *     sparse return descriptor
     * @param PaymentMethod $paymentMethod
     *     The PaymentMethod used for this Payment.
     * @param decimal $amount
     *     Amount to pay.
     * @param string $currency
     *     ISO 4217 currency code for amount. This must match the currency used for charges on the current invoice (if not specified, the AutoBill/Invoice currency will be used).
     * @param string $invoiceId
     *     ID of Invoice to make payment against. If not specified, the oldest unpaid invoice for this AutoBill will be selected for payment.
     * @param PaymentOverageDisposition $overageDisposition
     *     How to disposition amounts in excess of a required AutoBill payment amount. Defaults to applyToOldestInvoice if not specified
     * @param boolean $usePaymentMethodForFutureBilling
     *     Update the AutoBill to use the included PaymentMethod for future billing. Not Currently Supported
     * @param string $note
     *     Note regarding payment being made
     *
     * @return array - Elements:<ul>
     *         <li>boolean(0/1) '_completed'
     *         <li>int 'returnCode'
     *         <li>string 'returnString'
     *         <li>Update ResponseObject 'data'</ul>
     *
     * @internal UpdateResponse object 'data' contains:<ul>
     * <li>Return 'return' - Basic results of this request (success, failure, error, etc.)
     * <li>Transaction 'transaction' - The transaction generated by the payment attempt. This transaction must be inspected to assess the details of the payment attempt.
     * <li>TransactionAttemptSummary 'summary' - Summary of payment attempt</ul>
     */
    function makePayment($srd, $paymentMethod, $amount, $currency, $invoiceId, $overageDisposition, $usePaymentMethodForFutureBilling, $note) {
        $wsdl = VIN_SOAP_WSDL_PATH . '/AutoBill.wsdl';
        $auth = new Authentication();
        $input = array();
        $input['timeout'] = VIN_SOAP_TIMEOUT;
        $in_arr = array();
        $in_arr['auth'] = $auth;

        $in_arr['autobill'] = $this;
        $in_arr['srd'] = $srd;
        $in_arr['paymentMethod'] = $paymentMethod;
        $in_arr['amount'] = $amount;
        $in_arr['currency'] = $currency;
        $in_arr['invoiceId'] = $invoiceId;
        $in_arr['overageDisposition'] = $overageDisposition;
        $in_arr['usePaymentMethodForFutureBilling'] = $usePaymentMethodForFutureBilling;
        $in_arr['note'] = $note;

        $input['parameters'] = $in_arr;
        $soap = new CallClient();
        $ret = $soap->call($wsdl, 'AutoBill', 'makePayment', $input);
        return $ret;
    }
```

Reply:
I think that's default behavior
we could test that and than add a param maybe?

Reply:
could you email Vindicia to confirm that it does nothing and what the default behavior is?